src/txn.h

```diff
@@ -31,26 +36,31 @@ typedef enum {
 	TXN_ELEM_TXN_SIG,
 } txnElemType_e;
 
+typedef struct {
+    txnElemType_e elemType; // type of most-recently-seen element
+
+    uint8_t outVal[128];    // most-recently-seen currency value, in decimal
+    uint8_t valLen;         // length of outVal
+    uint8_t outAddr[77];    // most-recently-seen address
```

Comment:
if we store these in binary, we can save quite a bit of space: `outVal` could shrink to 16 bytes, and `outAddr` to 32.